wordpress-mobile/WordPress-Android
# Mobile stories block - part 4: error handling

This PR builds on top of https://github.com/wordpress-mobile/WordPress-Android/pull/13101

This contains all the work needed in order to provide error situation handling when saving or uploading any of the media related to a Story block.

![errorhandling2](https://user-images.githubusercontent.com/6597771/95992144-01ee8900-0e04-11eb-8eea-4bf3e7edd60e.gif)

To test:
CASE A: error in saves
1. apply this gist https://gist.github.com/mzorz/83b64b49d7a6160cc7beb7c3fd329d15 (this creates an exception the second time the service runs into an image slide to be saved (could be the same 1-slide story being saved more than once or a different slide on the same story)
2. run WPAndroid
3. create a story with 1 image slide and at least 1 view, it should be saved and published correctly
4. now go to the Posts list open the post on mobile gutenberg
5. select the story block, then tap again or tap on the edit icon button on the block to launch the Story composer
6. do edits like, move the added view around or add another view
7. tap on DONE, this should exit the composer and trigger the saving process
8. observe the process runs into an error and signals Gutenberg to show the failed overlay
9. tapping on the block should launch the story composer again, in error-wrangling mode
10. tap on RETRY to attempt a new save (this time it'll work)
11. now tap on DONE
12. observe the block gets updated on mobile gutenberg
13. if you want to see the changes reflected on your site, don't forget to tap on the UPDATE button on the editor as well.

CASE B: error in upload.
1. create a story with a few slides (3-4 so it takes enough time)
2. once its uploaded, edit it
3. tap on the story block and launch the story composer, now edit some views and tap DONE
4. the story will start uploading
5. turn airplane mode ON while uploads are still happening
6. observe the failed upload error overlay is shown
7. turn airplane mode oFF
8. tap on the story block failed overlay and observe the dialog appears "Couldn't upload file"
9. tap RETRY
10. observe the uploads resume and everything ends up correctly


Related PRs:

Gutenberg mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2708
Gutenberg: https://github.com/WordPress/gutenberg/pull/26008
Jetpack: https://github.com/Automattic/jetpack/pull/17458

PR submission checklist:
